### PR TITLE
Documentation of Ara

### PR DESCRIPTION
**Pull request by sharafat-10xEngineers**
This PR Adds the basic structure for Ara Documentation starting with Complete documentation for Vector Register File. Sphinx documentation generator is used to convert reStructuredText (.rst) files into html pages with readthedocs theme. CI support is added to generate documentation from source (rst files) and make them live on github pages. Following are the changes used to add support of above features:

### Added

- Add the basic structure of Ara documentation
- Add Vector Register File docs
- Add CI support to automatically generate docs for github pages

## Checklist

- [x] Automated tests pass
- [x] Changelog updated
- [x] Code style guideline is observed

Please check our [contributing guidelines](CONTRIBUTING.md) before opening a Pull Request.

### CODE REVIEW

**Review by mp-17 — COMMENTED** [review]
Thanks a lot for issuing this PR. This is a really good starting point for the documentation!

**Review by mp-17 — APPROVED** [review]

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Configuration file for the Sphinx documentation builder.
+#
+# For the full list of built-in configuration values, see the documentation:
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+
+project = 'Ara'
+copyright = '2022, Sharafat Hussain'
```
Comment (mp-17) [inline docs/source/conf.py:10]: We think there can be a problem with specifying a `copyright` here without any licensing information. 
Do you really want to copyright the documentation? If not, you can remove this line and we can merge. Of course, you would remain the author, and you can be acknowledged for it with the `author` line!

**Review by mp-17 — CHANGES_REQUESTED** [review]

File: docs/source/conf.py (hunk 1 above)
Comment (sharafat-10xEngineers) [reply docs/source/conf.py]: I agree. I have removed it.

File: docs/source/lane.rst [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+* As lanes are our processing elements so most computation constrained within one lane makes hardware implementation easy.
+
+* Removes the dependency of the execution units  on the number of lanes which makes Ara scalable.
+
+* As VRF is internal to the lane routing between VRF and lane execution logic is simplified.
+
+
+Division of VRF into Banks
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+As indicated earlier the portion of VRF inside a lane is further divided into eight 64-bit wide banks. Each bank is implemented using a single-ported SRAM memory. Being a single ported memory it has only one address port for both reads and writes. In steady state, under worst case conditions, 5 banks are accessed simultaneously in order to support predicated multiply-accumulate instructions (which require 3 source registers plus the mask register resulting in 4 reads, and one destination register -resulting in a single write).
```
[the remark this answers is not in the document]
Comment (sharafat-10xEngineers) [reply docs/source/lane.rst]: updated.

**Comment by mp-17** [comment]
LGTM now, thanks a lot.
@suehtamacv

**Comment by sharafat-10xEngineers** [comment]
Can you merge this PR now  @mp-17 @suehtamacv.

**Review by suehtamacv — APPROVED** [review]

**Pull request merged**